File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;
/*
 * 品牌接口
 * */
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.pojo.TbItem;
import entity.PageResult;

import java.util.List;
import java.util.Map;


public interface BrandService {
	public List<TbBrand> findAll();
	//分页
	public PageResult findPage(int pageNum, int pageSize);

	//增加
	public void add(TbBrand tbBrand);

	//根据id查询
    public TbBrand findOne(Long id);
    //修改
    public void update(TbBrand tbBrand);

    //删除
	public void delete(Long[] ids);

	//条件分页
	public PageResult findPage(TbBrand tbBrand ,int pageNum, int pageSize);

	//返回下拉列表数据
	public List<Map> selectOptionList();


}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    /*this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }*/

    this.search = function (page,size,searchEntity) {
        return $http.post('../brand/search.do?pageNum='+page +'&pageSize='+size,searchEntity);
    }
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }
    this.add = function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    this.update = function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    this.dele = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }
    //下拉列表数据
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    }
});